shell/renderer/content_settings_observer.cc

```diff
@@ -32,6 +43,24 @@ bool ContentSettingsObserver::AllowStorageAccessSync(StorageType storage_type) {
   return true;
 }
 
+bool ContentSettingsObserver::AllowReadFromClipboardSync() {
+  blink::WebLocalFrame* frame = render_frame()->GetWebFrame();
+  if (frame->GetSecurityOrigin().IsOpaque()) {
+    return false;
+  }
```

Comment:
Would this disallow this paste method on `about:blank`?

Reply:
Yeah given the security concerns we have with this feature, I am just allowing it for frames where users can reason with proper origin checks.

Reply:
I believe this is consistent with existing permission handler behavior. For cases where this should be checked, it's possible to grab `WebContents.opener`.

Reply:
Addressed in https://github.com/electron/electron/pull/45377/commits/68ac3e772e841f2d40c7dfca814f88f4a1457a92